File: Bai_26/main.py
#Cách 1
# Định nghĩa hàm để đếm số lần xuất hiện của chữ số k trong dãy A
def count_digit_occurrences(A, k):
    count = 0
    for number in A:
        count += str(number).count(str(k))
    return count
# Đọc số lượng bộ dữ liệu
T = int(input())
# Xử lý từng bộ dữ liệu
for _ in range(T):
    n, k = map(int, input().split())
    A = list(map(int, input().split()))
    # Gọi hàm để đếm số lần xuất hiện của chữ số k trong dãy A và in ra kết quả
    result = count_digit_occurrences(A, k)
    print(result)

#Cách 2
n,k = map(int, input().split())
A = input()
print(A.count(str(k)))

File: Bai_26/main.cpp
#include <iostream>
#include <vector>
using namespace std;

// Hàm để đếm số lần xuất hiện của chữ số k trong dãy A
int countDigitOccurrences(vector<int> A, int k) 
{
    int count = 0;
    for(int number : A) 
    {
        string numStr = to_string(number);
        for(char digit : numStr) 
        {
            if (digit - '0' == k) 
            {
                count++;
            }
        }
    }
    return count;
}

int main() 
{
    int T;
    cin >> T;

    for (int t = 0; t < T; ++t) 
    {
        int n, k;
        cin >> n >> k;

        vector<int> A(n);
        for (int i = 0; i < n; ++i) 
        {
           cin >> A[i];
        }

        // Gọi hàm để đếm số lần xuất hiện của chữ số k trong dãy A và in ra kết quả
        int result = countDigitOccurrences(A, k);
        cout << result << endl;
    }

    return 0;
}
